File: src/main/java/loxal/epvin/datastore/service/ResourceDAO.java
/*
 * Copyright 2012 Alexander Orlov <alexander.orlov@loxal.net>. All rights reserved.
 */

package loxal.epvin.datastore.service;

import com.googlecode.objectify.Objectify;
import com.googlecode.objectify.ObjectifyService;
import loxal.epvin.datastore.model.Resource;

import java.util.List;

/**
 * @author Alexander Orlov <alexander.orlov@loxal.net>
 */
public class ResourceDAO extends DAO {
    final Objectify ofy = ObjectifyService.begin();

    public void put(final Resource resource) {
        ofy.save().entity(resource).now();
    }

    public Resource get(final Long id) {
        return ofy.load().type(Resource.class).id(id).get();
    }

    public List<Resource> retrieve() {
        return ofy.load().type(Resource.class).list();
    }
}

File: src/main/kotlin/loxal/epvin/datastore/service/ResourceDAO.kt
package loxal.epvin.datastore.service

import loxal.epvin.datastore.model.Resource
import com.googlecode.objectify.ObjectifyService


public class ResourceDAO : DAO() {
    val ofy = ObjectifyService.ofy()

    public fun put(resource: Resource) {
        ofy.save().entity<Resource>(resource).now()
    }

    public fun get(id: Long?): Resource {
        return ofy.load().type<Resource>(Resource::class.java).id(id!!).now()
    }

    public fun retrieve(): List<Resource> {
        return ofy.load().type<Resource>(Resource::class.java).list()
    }
}